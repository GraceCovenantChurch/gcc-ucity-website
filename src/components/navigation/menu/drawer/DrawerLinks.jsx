import React from "react";
import PropTypes from "prop-types";
import clsx from "clsx";

import { makeStyles } from "@material-ui/core/styles";
import ListItem from "@material-ui/core/ListItem";
import ListItemText from "@material-ui/core/ListItemText";
import Collapse from "@material-ui/core/Collapse";
import ExpandLess from "@material-ui/icons/ExpandLess";
import ExpandMore from "@material-ui/icons/ExpandMore";

import Link from "components/link/Link";

const useStyles = makeStyles((theme) => ({
  menuText: {},
  collapse: { paddingLeft: "1em" },
  menuSubtitle: {
    // fontWeight: 700,
    // fontSize: "large",
  },
  title: {
    flexGrow: 1,
  },
  logo: {},
}));

const DrawerLinks = (props) => {
  const classes = useStyles();
  const [open, setOpen] = React.useState(false);

  const handleClick = () => {
    setOpen(!open);
  };

  return (
    <React.Fragment>
      {props.data.map((element, index) => {
        if (typeof element.children === "object") {
          return (
            <div className={""} key={index}>
              <ListItem
                button
                onClick={handleClick}
                className={clsx(classes.menuSubtitle, classes.menuText)}
              >
                {element.title}
                {open ? <ExpandLess /> : <ExpandMore />}
              </ListItem>
              <Collapse
                className={classes.collapse}
                in={open}
                timeout="auto"
                unmountOnExit
              >
                <DrawerLinks
                  data={element.children}
                  toggleDrawer={(flags) => props.toggleDrawer(false)}
                />
              </Collapse>
            </div>
          );
        } else {
          return (
            <div
              className={clsx(classes.list, [classes.fullList])}
              role="presentation"
              onClick={props.toggleDrawer(false)}
              onKeyDown={props.toggleDrawer(false)}
              key={index}
            >
              <ListItem button component={Link} to={element.children}>
                <ListItemText
                  className={classes.menuText}
                  primary={element.title}
                />
              </ListItem>
            </div>
          );
        }
      })}
    </React.Fragment>
  );
};

DrawerLinks.propTypes = {
  data: PropTypes.arrayOf(
    PropTypes.shape({
      title: PropTypes.string.isRequired,
      children: PropTypes.any,
    })
  ),
  toggleDrawer: PropTypes.func.isRequired,
};
export default DrawerLinks;
