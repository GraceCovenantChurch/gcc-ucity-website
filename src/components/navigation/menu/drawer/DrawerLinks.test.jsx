import React from "react";
import "../../../../__mocks__/matchMedia";
import { Context as ResponsiveContext } from "react-responsive";
import { render } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";

import DrawerLinks from "./DrawerLinks";

const menuOptions = [
  {
    title: "About",
    type: "dropdown",
    children: [{ title: "Welcome", type: "link", children: "/welcome" }],
  },
  { title: "Family Groups", type: "link", children: "/familygroup" },
];

const toggleDrawer = jest.fn();

describe("DrawerLinks", () => {
  test("matches the snapshot", () => {
    const { container: mobile } = render(
      <Router>
        <ResponsiveContext.Provider value={{ width: 300 }}>
          <DrawerLinks
            toggleDrawer={(flag) => toggleDrawer(flag)}
            data={menuOptions}
          />
        </ResponsiveContext.Provider>
      </Router>
    );
    expect(mobile).toMatchSnapshot();

    const { container: desktop } = render(
      <Router>
        <ResponsiveContext.Provider value={{ width: 1000 }}>
          <DrawerLinks
            toggleDrawer={(flag) => toggleDrawer(flag)}
            data={menuOptions}
          />
        </ResponsiveContext.Provider>
      </Router>
    );
    expect(desktop).toMatchSnapshot();
  });
});
