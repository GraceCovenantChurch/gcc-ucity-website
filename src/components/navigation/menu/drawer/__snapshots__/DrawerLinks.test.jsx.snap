// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DrawerLinks matches the snapshot 1`] = `
<div>
  <a
    aria-disabled="false"
    class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
    href="/"
    role="button"
    tabindex="0"
  >
    <div
      class="MuiListItemText-root makeStyles-menuText-1"
    >
      <span
        class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
      >
        Home
      </span>
    </div>
    <span
      class="MuiTouchRipple-root"
    />
  </a>
  <div
    class=""
  >
    <div
      aria-disabled="false"
      class="MuiButtonBase-root MuiListItem-root makeStyles-menuSubtitle-3 makeStyles-menuText-1 MuiListItem-gutters MuiListItem-button"
      role="button"
      tabindex="0"
    >
      About
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
        />
      </svg>
      <span
        class="MuiTouchRipple-root"
      />
    </div>
  </div>
  <div
    class=""
    role="presentation"
  >
    <a
      aria-disabled="false"
      class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
      href="/familygroup"
      role="button"
      tabindex="0"
    >
      <div
        class="MuiListItemText-root makeStyles-menuText-1"
      >
        <span
          class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
        >
          Family Groups
        </span>
      </div>
      <span
        class="MuiTouchRipple-root"
      />
    </a>
  </div>
</div>
`;

exports[`DrawerLinks matches the snapshot 2`] = `
<div>
  <a
    aria-disabled="false"
    class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
    href="/"
    role="button"
    tabindex="0"
  >
    <div
      class="MuiListItemText-root makeStyles-menuText-1"
    >
      <span
        class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
      >
        Home
      </span>
    </div>
    <span
      class="MuiTouchRipple-root"
    />
  </a>
  <div
    class=""
  >
    <div
      aria-disabled="false"
      class="MuiButtonBase-root MuiListItem-root makeStyles-menuSubtitle-3 makeStyles-menuText-1 MuiListItem-gutters MuiListItem-button"
      role="button"
      tabindex="0"
    >
      About
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
        />
      </svg>
      <span
        class="MuiTouchRipple-root"
      />
    </div>
  </div>
  <div
    class=""
    role="presentation"
  >
    <a
      aria-disabled="false"
      class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
      href="/familygroup"
      role="button"
      tabindex="0"
    >
      <div
        class="MuiListItemText-root makeStyles-menuText-1"
      >
        <span
          class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
        >
          Family Groups
        </span>
      </div>
      <span
        class="MuiTouchRipple-root"
      />
    </a>
  </div>
</div>
`;
